cli/purge.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+func MakePurgeCommand() *cobra.Command {
+	var force bool
+	var cmd = &cobra.Command{
+		Use:   "purge",
+		Short: "Delete all persisted DefraDB data",
+		Long: `Delete all persisted DefraDB data.
+WARNING this operation will delete all data and cannot be reversed.`,
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if err := setContextRootDir(cmd); err != nil {
+				return err
+			}
+			return setContextConfig(cmd)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if !force {
+				return ErrPurgeForceFlagRequired
+			}
+			cfg := mustGetContextConfig(cmd)
+			return os.RemoveAll(cfg.GetString("datastore.badger.path"))
+		},
+	}
+	cmd.Flags().BoolVarP(&force, "force", "f", false, "Must be set for the operation to run")
```

Comment:
todo: I don't think  requiring `-f` is sufficient protection against users nuking a production database, I can't remember the details but I thought we agreed on something slightly safer when we discussed this as a team?

Reply:
I only remember the force flag from the discussion. As long as its non interactive I'm open to changing it to anything else.

Reply:
I think we had discussed the possibility of having a `dev` flag on startup with only `dev` instances capable of calling `purge`.  If we do this, I think we don't even need `force`. Also, having a `dev` flag will allow for other commands to be added to `dev` only in the future. `dev` could by default require no password for the keyring for example. This could all be part of a strategy to get new devs to the "fun part" asap.

Reply:
> I think we had discussed the possibility of having a `dev` flag on startup with only `dev` instances capable of calling `purge`. If we do this, I think we don't even need `force`. Also, having a `dev` flag will allow for other commands to be added to `dev` only in the future. `dev` could by default require no password for the keyring for example. This could all be part of a strategy to get new devs to the "fun part" asap.

I don't remember this, but I like it - is kind of like a build flag without the hassle of multiple builds, and like you said it scales to other situations very nicely too